server/api/contest_handler.py

```diff
@@ -36,19 +36,32 @@ def create_contest():
 def get_all_contests():
     # Do any contests exist?
     try:
-        all_contests = Contest.query.all()
+        all_contests = db.session.query(User, Contest).outerjoin(Contest, Contest.contest_creater == User.id).all()
         if not bool(all_contests):
-            raise Exception
-    except Exception:
+            raise Exception("no contest")
+    except Exception as e:
+        print(e)
         return jsonify("No contests listed")
     # Return all contests
     else:
-        dictionary = {}
-        counter = 0
-        for contest in all_contests:
-            dictionary["contest_{contest_number}".format(contest_number = counter)] = contest.__dict__
-            counter += 1
-        return json.dumps(dictionary, default=str)
+        lst = []
+        ind = 0
+        for pair in all_contests:
+            user, contest = pair
+            if (user is not None) and (contest is not None):
+                lst.append({
+                    "ind": ind,

```

Comment:
ind is used for index? If so, we don't necessarily need to return this or keep track of it. a better form of ID would be the contest ID